tiatoolbox/cli/slide_info.py

```diff
@@ -20,54 +20,57 @@
 
 """Command line interface for slide_info."""
 import pathlib
-import sys
-
-import click
 
 from tiatoolbox import utils, wsicore
+from tiatoolbox.cli.common import (
+    TIAToolboxCLI,
+    cli_file_type,
+    cli_img_input,
+    cli_mode,
+    cli_output_path,
+    cli_verbose,
+    no_input_message,
+    prepare_file_dir_cli,
+)
 
-
-@click.group()
-def main():  # pragma: no cover
-    """Define slide_info click group."""
-    return 0
+slide_info_cli = TIAToolboxCLI()
```

Comment:
Looks like you create this object in each file. Is that the intended way from click? I am not sure because I do not know the library that well. It looks like you want this to be a singleton which inheriting from Group might make it but I am not sure.

Reply:
I am creating these for now but will address this in a separate PR.
The plan is to create an abstract class with subclasses for each command which invokes only if called with lazy imports.

Reply:
Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/307/commits/c19e3cc7d4ac22cd26b37ca40b0790edf71d27ce